npc/custom/item_options.txt

```diff
@@ -0,0 +1,234 @@
+		mes(.@name$);
+		mes("Alright then, catch you later.");
+		close;
+	}
+	// Build the Menu.
+	setarray .@position$[1],"Head","Body","Left Hand","Right Hand","Robe","Shoes","Accessory 1","Accessory 2","Head 2","Head 3";
+	.@menu$ = "";
+	for (.@i = 1; .@i<=10; ++.@i)
+		.@menu$ += ((getequipisequiped(.@i))?getequipname(.@i):.@position$[.@i]+"-[Not equipped]")+":";
+	// Select the part.
+	.@equip_index = select(.@menu$);
+	
+	// Check if it's worn.
+	if (!getequipisequiped(.@equip_index)) {
+		mes(.@name$);
+		mes("Brah, your item ain't equipped.");
+		close;
+	// Check if it allows options
+	} else if (!getequipisenableopt(.@equip_index)) {
+		mes(.@name$);
+		mes("Brah, this equip can't have options.");
+		close;
+	// Check if equipment is identified.
+	} else if (!getequipisidentify(.@equip_index)) {
+		mes(.@name$);
+		mes("Brah, this equip ain't identified.");
+		close;
+	}
+	
+	// Present with a list of currently enfused options.
+	do {
+		.@used = false; // unset flag for used slot.
+		.@menu$ = "";
+		// Build the menu of current options.
+		for (.@i = 0; .@i < MAX_ITEM_OPTIONS; .@i++) {
+			// call getequipoption(<equip_index>, <type>, <slot>);
+			// if the return is <0, it's a script error.
+			// if the return is 0, the slot is empty.
+			// if the return is >0, the slot is unavailable.
+			.@opt = getequipoption(.@equip_index, IT_OPT_INDEX, .@i);
+			if (.@opt > 0)
+				.@menu$ += (.@i+1)+") "+.options$[.@opt-1]+":";
+			else
+				.@menu$ += (.@i+1)+") ^999999Empty^000000"+":";
+		}
+		// Option Slot is the actual option slot 0-MAX_ITEM_OPTIONS (@see mmo.h)
+		.@option_slot = select(.@menu$)-1;
+		
+		// Check for used slot and request user action if found.
+		if (getequipoption(.@equip_index, IT_OPT_INDEX, .@option_slot) > 0) {
+			mes(.@name$);
+			mes("This slot is already used up!");
+			next;
+			if (select("^990000Override the slot.^000000", "Choose again.") == 2) {
+				.@used = true;
+			}
+		}
+	} while (.@used);
+	
+	// Present with a list of available bonuses.
+	mes(.@name$);
+	mes("Which of the following item bonuses would you like to add to this item?");
+	next;
+	// Build the Options!
+	.@menu$ = "";
+	for (.@i = 0; .@i < getarraysize(.options$); .@i++)
+		.@menu$ += (.@i+1)+") "+.options$[.@i]+":";
+		
+	do {
+		// Select the options!
+		.@option_variable = select(.@menu$);
+		next;
+		mes(.@name$);
+		mes("You chose ^009900"+.options$[.@option_variable-1]+"^000000!");
+		next;
+	} while (select("Fo Shizzle.", "I'ma re-evaluate, brah.") == 2);
+	
+	// Select a random bonus.
+	if (.enable_random_bonus) {
+		.@value = rand(.maximum_bonus_amount);
+	} else {
+		do {
+			mes(.@name$);
+			mes("Please input the bonus amount of ^009900"+.options$[.@option_variable-1]+"^000000 you want to add!");
+			mes("(Min: "+.minimum_bonus_amount+", Max: "+.maximum_bonus_amount+")");
+			input .@value;
+			next;
+		} while (.@value < .minimum_bonus_amount || .@value == 0 || .@value > .maximum_bonus_amount);
```

Comment:
extra condition checking can be avoided if you applied the `input(<variable>,<min>,<max);`